### PR TITLE
Read model weights 

### PR DESCRIPTION
**Pull request by shavit**
Read model weights from safetensors.

Related https://github.com/huggingface/swift-transformers/pull/86

### CODE REVIEW

**Comment by shavit** [comment]
Weights are now accessible independently of configurations for flexibility and without MLModel or conversion.

**Comment by pcuenca** [comment]
That's very cool @shavit! May I ask what use case you have in mind for this?

**Comment by shavit** [comment]
Yes
  * In the first commit I added weights property to `Configurations`, but since `LanguageModel` requires a model, I separated the weights initialization.
   * The weights can go into the `Embedding` module instead, since the language models in this module are tied to CoreML, and don't need this at the moment.
   * The `Safetensor` and others can save updated weights into a file.

The plan is to use models directly from local files or the hub, and maybe support more types like mlx/numpy.

File: Tests/TensorUtilsTests/WeightsTests.swift [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+    let downloadDestination: URL = {
+        FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first!.appending(component: "huggingface-tests")
+    }()
+
+    var hubApi: HubApi { HubApi(downloadBase: downloadDestination) }
+
+    func testLoadWeightsFromFileURL() async throws {
+        let repo = "google/bert_uncased_L-2_H-128_A-2"
+        let modelDir = try await hubApi.snapshot(from: repo)
```
Comment (pcuenca) [inline Tests/TensorUtilsTests/WeightsTests.swift:15]: I'd suggest you use [matching globs](https://github.com/huggingface/swift-transformers/blob/37e234e1ee4f2ef9c21c3d5cd592b5ec63d879af/Tests/HubTests/HubApiTests.swift#L188) to only download the two files we need and avoid additional downloads. There's a [version](https://github.com/huggingface/swift-transformers/blob/37e234e1ee4f2ef9c21c3d5cd592b5ec63d879af/Sources/Hub/HubApi.swift#L204) that supports multiple patterns.

**Review by pcuenca — APPROVED** [review]
Very cool work, thanks a lot for contributing! I only have one suggestion regarding one of the tests.

**Review by pcuenca — APPROVED** [review]
Ready to merge, in my opinion! Are you planning any more tweaks?

**Comment by shavit** [comment]
No, that's it.

**Comment by pcuenca** [comment]
Thanks a lot @shavit!

**Pull request merged**